src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/UpdateFinancialTransactionCategoryIT.java

```diff
@@ -0,0 +1,241 @@
+
+    ResultActions resultActions = mockMvc.perform(
+      MockMvcRequestBuilders.patch(ENDPOINT_CATEGORIES_PATCH, financialTransactionCategory.getId())
+        .header(BaseIntegrationTestIT.AUTHORIZATION, BaseIntegrationTestIT.BEARER + accessToken)
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(objectMapper.writeValueAsString(financialTransactionCategoryUpdateDTO))
+        .accept(MediaType.APPLICATION_JSON)
+    );
+
+    resultActions.andExpectAll(
+      status().isBadRequest(),
+      jsonPath("$.status").value(ErrorCode.TEA003.getBusinessStatus()),
+      jsonPath("$.message").value(ErrorCode.TEA003.getBusinessMessage()),
+      jsonPath("$.statusCode").value(ErrorCode.TEA003.getBusinessStatusCode())
+    );
+  }
+
+  @DisplayName("Should not update FT category when category name contains invalid characters")
+  @Test
+  public void testUpdateFTCategoryFailure_whenCategoryNameContainsInvalidCharacters_thenReturnBadRequest() throws Exception {
+    User user = createUser("user@domain.server.com");
+    String accessToken = userService.createAccessToken(user);
+
+    FinancialTransactionCategory financialTransactionCategory = createFinancialTransactionCategory(user);
+    FinancialTransactionCategoryUpdateDTO financialTransactionCategoryUpdateDTO =
+      new FinancialTransactionCategoryUpdateDTO("!@`$%^&*()_+|-=[];',./{}:<", FinancialTransactionType.EXPENSE);
+
+    ResultActions resultActions = mockMvc.perform(
+      MockMvcRequestBuilders.patch(ENDPOINT_CATEGORIES_PATCH, financialTransactionCategory.getId())
+        .header(BaseIntegrationTestIT.AUTHORIZATION, BaseIntegrationTestIT.BEARER + accessToken)
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(objectMapper.writeValueAsString(financialTransactionCategoryUpdateDTO))
+        .accept(MediaType.APPLICATION_JSON)
+    );
+
+    resultActions.andExpectAll(
+      status().is5xxServerError(),
+      jsonPath("$.status").value(ErrorCode.TEA004.getBusinessStatus()),
+      jsonPath("$.message").value(ErrorCode.TEA004.getBusinessMessage()),
+      jsonPath("$.statusCode").value(ErrorCode.TEA004.getBusinessStatusCode())
+    );
+  }
+
+  @DisplayName("Should not update FT category when category type is invalid")
+  @Test
+  public void testUpdateFTCategoryFailure_whenCategoryTypeIsInvalid_thenReturnBadRequest() throws Exception {
+    User user = createUser("user@domain.server.com");
+    String accessToken = userService.createAccessToken(user);
+
+    FinancialTransactionCategory financialTransactionCategory = createFinancialTransactionCategory(user);
+
+    ResultActions resultActions = mockMvc.perform(
+      MockMvcRequestBuilders.patch(ENDPOINT_CATEGORIES_PATCH, financialTransactionCategory.getId())
+        .header(BaseIntegrationTestIT.AUTHORIZATION, BaseIntegrationTestIT.BEARER + accessToken)
+        .contentType(MediaType.APPLICATION_JSON)
+        .content("{\"name\": \"TEST\", \"type\": \"INVALID_CATEGORY_TYPE\"}")
```

Comment:
To generate JSON value from a few String values we can use a private method, i.e:
```java
public String generateJSON(String value1, String value2) {
        JsonObject jsonObject = new JsonObject();
        jsonObject.addProperty("key1", value1);
        jsonObject.addProperty("key2", value2);
   
        ObjectMapper objectMapper = new ObjectMapper();
        try {
            return objectMapper.writeValueAsString(jsonObject);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        }

        return null;
    }
```

**Needed dependencies in _pom.xml_:**
```xml
<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-databind</artifactId>
    <version>2.15.2</version>
</dependency>
```

**Needed imports in integration test:**
```java
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
```

Reply:
p.s. please test it before commit & push

Reply:
We cannot add this because we have already used Jackson library for ObjectMapper. When we add library this create an error. More details: https://stackoverflow.com/questions/61169128/could-not-write-json-jsonobject-nested-exception-is-com-fasterxml-jackson-data